### PR TITLE
chore(clerk-js): Add test coverage for memoizeStateListenerCallback

### PR DESCRIPTION
**Pull request by jacekradko**
## Description

Adding base test coverage for memoizeStateListenerCallback. Primarily added this when verifying we are handling the `null` to `undefined` and vice-versa scenarios

## Checklist

- [ ] `pnpm test` runs as expected.
- [ ] `pnpm build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerk/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

## Summary by CodeRabbit

* **Tests**
  * Expanded test coverage for state-listener memoization, adding comprehensive edge-case validation including null vs. undefined transitions and repeated-call behavior.
* **Chores**
  * Added a changeset placeholder file (no runtime or API changes).

### CODE REVIEW

File: .changeset/slimy-maps-read.md [hunk 1]
```diff
@@ -0,0 +1,2 @@
+---
+---
```
[the remark this answers is not in the document]
Comment (jacekradko) [reply .changeset/slimy-maps-read.md:2]: We accept empty changesets as a signal that no bump is intended
